### ISSUE TITLE
Failed installation

### ISSUE THREAD

**Issue by greymonroe** [RcppCore/RcppArmadillo#249]
I am trying to update the RcppArmadillo package. 

```
Installing 1 packages: RcppArmadillo

  There is a binary version available but the source version is later:
                   binary      source needs_compilation
RcppArmadillo 0.9.200.7.0 0.9.200.7.1              TRUE

Do you want to install from sources the package which needs compilation? (Yes/no/cancel) 

```

Select Yes...

```
* installing *source* package ‘RcppArmadillo’ ...
** package ‘RcppArmadillo’ successfully unpacked and MD5 sums checked
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether clang++ accepts -g... yes
checking how to run the C++ preprocessor... clang++ -E
checking whether we are using the GNU C++ compiler... (cached) yes
checking whether clang++ accepts -g... (cached) yes
checking whether g++ version is sufficient... almost
configure: WARNING: Compiler self-identifies as being compliant with GNUC extensions but is not g++.
checking for macOS... found
checking for macOS Apple compiler... found
configure: WARNING: OpenMP unavailable and turned off.
checking LAPACK_LIBS... R-supplied partial LAPACK found
configure: WARNING: Some complex-valued LAPACK functions may not be available
configure: creating ./config.status
config.status: creating inst/include/RcppArmadilloConfigGenerated.h
config.status: creating src/Makevars
** libs
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I"/Library/Frameworks/R.framework/Versions/3.5/Resources/library/Rcpp/include" -I/usr/local/include  -I../inst/include  -fPIC  -Wall -g -O2 -c RcppArmadillo.cpp -o RcppArmadillo.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I"/Library/Frameworks/R.framework/Versions/3.5/Resources/library/Rcpp/include" -I/usr/local/include  -I../inst/include  -fPIC  -Wall -g -O2 -c RcppExports.cpp -o RcppExports.o
clang++ -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I"/Library/Frameworks/R.framework/Versions/3.5/Resources/library/Rcpp/include" -I/usr/local/include  -I../inst/include  -fPIC  -Wall -g -O2 -c fastLm.cpp -o fastLm.o
clang++ -std=gnu++11 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/usr/local/lib -o RcppArmadillo.so RcppArmadillo.o RcppExports.o fastLm.o -L/Library/Frameworks/R.framework/Resources/lib -lRlapack -L/Library/Frameworks/R.framework/Resources/lib -lRblas -L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin15/6.1.0 -L/usr/local/gfortran/lib -lgfortran -lquadmath -lm -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
ld: warning: directory not found for option '-L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin15/6.1.0'
ld: warning: directory not found for option '-L/usr/local/gfortran/lib'
ld: library not found for -lgfortran
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [RcppArmadillo.so] Error 1
ERROR: compilation failed for package ‘RcppArmadillo’
* removing ‘/Library/Frameworks/R.framework/Versions/3.5/Resources/library/RcppArmadillo’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/3.5/Resources/library/RcppArmadillo’
Error in i.p(...) : 
  (converted from warning) installation of package ‘RcppArmadillo’ had non-zero exit status
```

**Comment by eddelbuettel** [comment]
> ld: library not found for -lgfortran

**Comment by jananiravi** [comment]
I see similar errors after recently updating my R and gcc versions (incl. gfortran).
Gist of warnings: https://gist.github.com/jananiravi/2dc460fa3c1fd6c57ec6e58cdeaa293e
Gist of my RStudio environment: https://gist.github.com/jananiravi/1136d342ee2b837eed3d6c6dccfb8f23

Thanks for your help.

**Comment by eddelbuettel** [comment]
What OS?

**Comment by eddelbuettel** [comment]
For macOS you _must_ follow the detailed instructions provided _e.g._ [here by @coatless](https://thecoatlessprofessor.com/programming/r-compiler-tools-for-rcpp-on-macos/) -- and I _do not use macOS myself_ so I am not sure if I pointed you to the right page (among many).  Also see [the Rcpp FAQ](https://cloud.r-project.org/web/packages/Rcpp/vignettes/Rcpp-FAQ.pdf). 

The toolchain is a given for us.  We don't influence it.  On macOS, you either use it, or you get tears. Sorry -- but our hands are as tied as yours here.

**Comment by jananiravi** [comment]
Thank you, @eddelbuettel. The resources by @coatless and you were quite helpful considering my naivete in clang/compilation theory. Yes, these are on macOS w/ the latest dev version of RStudio (`v1.2.1114`) and R (`v3.6.0`).

I do have the latest version of llvm installed, though using homebrew and that includes fairly recent versions of clang & clang++ (8.0). I have, therefore, not reinstalled clang once again.
```
clang version 8.0.0 (tags/RELEASE_800/final)
Target: x86_64-apple-darwin18.5.0
Thread model: posix
InstalledDir: /usr/local/opt/llvm/bin
```

I am not sure if I am encountering [my issue](https://github.com/thomasp85/FindMyFriends/issues/17) because of the way I've set my compilation defaults in my [make files](https://gist.github.com/jananiravi/1702e9bc1be61b4602a3e2aa285b3238). I am still figuring that out.
So, if you or @coatless could share some insight on what these `clang`, `cdhit` or `pansim` warnings mean, that would be most helpful. I understand that these are just warnings and that the package is still installed. I want to make sure I'm not missing any early warning signs of the way I've set things up.

Thanks for your help!
(_If I figure something out in the meantime, I will post an update here for the benefit of any others who are encountering these errors._)

**Comment by cripenam** [comment]
Worked for me after installing the Developers version:

install.packages("https://cran.r-project.org/src/contrib/Archive/RcppArmadillo/RcppArmadillo_0.9.900.3.0.tar.gz", repos=NULL, type="source")

**Comment by eddelbuettel** [comment]
As I wrote a minute ago over in #71:

> I am sorry but what value does that message add? We had three+ more releases since so 0.9.900.3.0 is outdated.

Same question here: we had four releases since. What value are you adding by pointing to an old release?

*Both* #71 and this #249 appear due to be caused by a lack of a Fortran compiler. No version, old or new, will fix this.

**Comment by cripenam** [comment]
What value? Sorry I didn't mean to offend, just a confused and naive user here as well. 
I tried to set up R  in a new MacOS big sur and run into the same mistake as some people in 2015 and 2019. 
Tried different ways and trouble shoot but was un-relatable (that's the value for a new user like me when i see post as: try this, try that). 
I got an advice form somebody after hours of just going in circles trying to compile a package. his advice was to install a dev. version, i picked one at random and it worked. 

Outdated? probably, worked? yes, why? idk, should i post how it got magically fixed in my pc? why not. 

Thanks for the hard work, just a student jumping into R in here.
